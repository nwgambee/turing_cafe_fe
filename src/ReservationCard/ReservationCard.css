.res-card {
  border: 2px solid teal;
  border-radius: 4px;
  background-color: white;
  color: teal;
  padding: 10px;
  margin: 3px;
  height: 20%;
  width: 25%;
}

.remove-res-btn {
  color: white;
  background-color: teal;
  font-size: 1em;
  padding: 4px;
  border-radius: 3px;
  cursor: pointer;
}
