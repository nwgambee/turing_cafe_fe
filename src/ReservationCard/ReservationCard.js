import React from 'react';

const ReservationCard  = () => {
console.log('res cards here ');
  return (
    <section className='res-card'>
    </section>
  )
}

export default ReservationCard;
