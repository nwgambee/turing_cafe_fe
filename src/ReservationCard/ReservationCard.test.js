import React from 'react';
import ReactDOM from 'react-dom';
import ReservationCard from './ReservationCard';
import {shallow} from 'enzyme';

it('should match the snapshot', () => {
  const wrapper = shallow(<ReservationCard
    name='Noah'
    id={123}
    date='12/13'
    time='3:45'
    number={4}
    cancelReservation={jest.fn()}
    />)
  expect(wrapper).toMatchSnapshot();
})

it.skip('should invoke cancelReservation when remove-res-btn is clicked', () => {
  const wrapper = shallow(<ReservationCard
    name='Noah'
    id={123}
    date='12/13'
    time='3:45'
    number={4}
    cancelReservation={jest.fn()}
    />)
  wrapper.instance().cancelReservation = jest.fn();
  wrapper.instance().forceUpdate();
  wrapper.find('.remove-res-btn').simulate('click')
  expect(wrapper.instance().cancelReservation).toHaveBeenCalled();
})
