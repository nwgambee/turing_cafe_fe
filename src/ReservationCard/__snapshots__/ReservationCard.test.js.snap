// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section
  className="res-card"
>
  <h3>
    Noah
  </h3>
  <p>
    12/13
  </p>
  <p>
    3:45
  </p>
  <p>
    4
     guests
  </p>
  <button
    className="remove-res-btn"
    onClick={[Function]}
  >
    Cancel
  </button>
</section>
`;
